standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/security/TokenStoreDelegationTokenSecretManager.java

```diff
@@ -105,7 +105,11 @@ public byte[] retrievePassword(DelegationTokenIdentifier identifier) throws Inva
       if (info == null) {
           throw new InvalidToken("token expired or does not exist: " + identifier);
       }
-      renewIfRequired(System.currentTimeMillis(), identifier, info);
+      try {
+        renewIfRequired(identifier, info);
+      } catch (InvalidToken e) {
+        LOGGER.warn("Failed to renew token: " + identifier, e);
```

Comment:
If the renew token fails, will the user's job fail?

Reply:
No, ongoing user's job will not fail as it does not require the token. There are 2 cases here:
1. If current time < renewal time : In this case it will never try to go for renewal.
2. current time >= renewal time : in this case it will attempt for token renewal. The renewal here is failsafe in case if the session ever requires the delegation token later in it's lifeline.

This will have no impact on the user's job.